### DISCUSSION TITLE
Is it mandatory to commit the flux_install (terraform) generated manifests

### DISCUSSION THREAD

**Discussion by Dotisandhyarani** [fluxcd/flux2#3143] [General] [answered]
Hi,

Auto deploying applications using flux on EKS.
I see many examples , the flux generated manfiest files are being commited to the git repo. Is it mandatory to commit those files ?

In my case, I'm using terraform and applying the flux manifests on EKS cluster and everything works fine even without committing flux generated manifests to the git repo, but I do apply the manifests to EKS cluster (using tf `kubectl_manifest`)

So, I'm guessing its not required to commit the manifests to git repo, just applying the manifest works

I am applying with kubectl_manifest 

```
resource "kubectl_manifest" "flux_install" {
  depends_on = [ kubernetes_namespace.flux_system ]
  for_each   = { for v in local.install : lower(join("/", compact([v.data.apiVersion, v.data.kind, lookup(v.data.metadata, "namespace", ""), v.data.metadata.name]))) => v.content }
  yaml_body  = each.value
}
```

Please advice,
Thanks!

**Comment by kingdonb** [comment, marked as answer]
It is not mandatory but it is highly recommended.

Flux managing itself via GitOps is kind of central to getting users to understand Flux's intended use. The terraform provider docs are made to mimic the form of a bootstrap, the central concept in Getting Started guide that allows Flux to manage itself. This means that on the cluster, it's not necessary to use Terraform to upgrade Flux once it has been installed, you can just terraform up your clusters once, and do your upgrades inside the clusters via GitOps after.

In practice you probably planned on running Terraform apply again later on, maybe this sets up a conflict for you, because you don't necessarily know where Flux is managed from – GitOps or Terraform – but we think that Terraform is not the best way to manage what runs inside your cluster. (That's what you will use Flux for!)

In the extreme ideal case, everything on the cluster is managed by Flux, the clusters themselves are managed by Flux, and external resources which support the cluster and app lifecycles are also managed on the cluster with an operator by Flux. In some cases things like CNI and control plane cannot be part of Flux's config and are managed externally. So to manage everything but Flux itself... well!

If you don't care about doing GitOps with consistency, then you can absolutely skip the commit step and just run `flux install`. It sounds like that's basically what you're doing but with TF. If you want to have a record of what was installed when, and manage Flux upgrading via Git, then you should commit the manifests into the repository and let Flux managing them be the only way that change gets onto the cluster, for consistency and auditability.

With Terraform, I think there's nothing stopping you from only applying the Flux manifests to the cluster and not committing them. I am not much of a Terraform user myself yet, so I don't know if in practice this might not be such a hard problem, but I think that Terraform has problems managing what's inside of the cluster, issues where the state can get inconsistent. I think it's generally used to "fire and forget" clusters when used together with Flux, in other words, Terraform only hits the cluster once when it's created (and again when it's being destroyed.) Let me know if I'm off-base about this please! Good to hear it from an actual Terraform user, I'm going from what I have heard.

**Reply by Dotisandhyarani** [reply]
Thanks for the response.

The only problem to commit the manifests to Bitbucket,  TF does not have  better resource support to commit the files to Bitbucket unlike GitHub. 
But there is a workaround, I had to clone the repo locally first and commit the files to Bitbucket.
